### ISSUE TITLE
Browser support

### ISSUE THREAD

**Issue by ghost** [diasurgical/DevilutionX#228]
Add a browser target for DeviltuionX via webasm.

Current status is that we can build for webasm, though the event model does not work with the webasm port of SDL as there can only be one event loop and we have about 5 atm.

**Comment by qndel** [comment]
It doesn't build as php either, what a garbage :(

**Comment by megamindbrian** [comment]
I like PHP, there is a babel-preset so after the convert to JS, I could attempt that too.

**Comment by megamindbrian** [comment]
Is the "render" branch planning on being merged soon? Just wondering if I should go off master or render to avoid merge issues later?

**Comment by qndel** [comment]
I'd vote for render

**Comment by AJenbo** [comment]
@briancullinan did you get time to work on this?

**Comment by briancullinan** [comment]
@AJenbo  Same problem I ran in to QuakeJS. Someone got excited and included Devilution in an emscripten build instead of the opposite and including emscripten in the Devilution build. 
https://github.com/d07RiV/diabloweb

If you build it they will come.
I don't know enough about makefile/cmake to reformat this entire script, but it should be pretty upgradable. Someone else did it first.

**Comment by AJenbo** [comment]
@briancullinan so is this still something you intend to work on? If you join discord we can probably help with the changes that are needed for the build process.

**Comment by ThomasChr** [comment]
@AJenbo I do not think that something will hapen here anymore...
Also I think there is a repository for this?

https://github.com/d07RiV/diabloweb

**Comment by AJenbo** [comment]
The gap between diabloweb and devilutionx is so big at this point that they barely compare. I would still like there to be build target for browsers in DevilutionX. But I'm also willing to close this issue as no on is currently working on it it seams.

**Comment by sskras** [comment]
@AJenbo commented [on Nov 11, 2021](#issuecomment-966402529):

> The gap between diabloweb and devilutionx is so big at this point that they barely compare. 

BTW, does this mean that I cannot reuse the diabloweb version of the single player save game in modern devilutionx ?

**Comment by StephenCWills** [comment]
I don't know much about diabloweb, but the savegame format in DevilutionX is compatible with vanilla Diablo.

**Comment by AJenbo** [comment]
@sskras savegames should be compatible. The comment is more in relation to reusing diabloweb code for getting DevilutonX working in a browser, since that is the topic of this issue.

**Comment by sskras** [comment]
@AJenbo: I tried loading it, and the devilutionx:

1, doesn't see the saved char/game and offers to created the new char.
2, (when I enter the same name), it crashes.

**Comment by AJenbo** [comment]
@sskras please don't post your questions in unrelated issues. This one is about adding browser support to DevilutionX.

If you have issues with save games make sure to attach them. Follow the issue template or use the chat.

**Comment by sskras** [comment]
I guess no one is capable to work on the web / browser support at the moment.

**Comment by AJenbo** [comment]
Thank you captain obvious

**Comment by Benman2785** [comment]
has something changed now?

what is the "how2" for webserver version?

**Comment by qndel** [comment]
> has something changed now?
> 
> what is the "how2" for webserver version?

no new comments indicate nothing has changed

**Comment by glebm** [comment]
The main blocker to this is #3776

**Comment by HoofedEar** [comment]
> Getting everything into a single event loop  


I get that this is a blocker for the web version, but is this something that is feasible? 
I'm very interested in a web version, but if a refactor putting everything into a single event loop is super massive, would that warrant it's own "web" fork?

**Comment by AJenbo** [comment]
No, cause you would have to do the refactor in the fork then. It's not that it will break things, it's that it has to be done.

**Comment by glebm** [comment]
> I get that this is a blocker for the web version, but is this something that is feasible?

It's a lot of work but not impossible. https://github.com/d07RiV/diabloweb did it.
